main.go

```diff
@@ -137,46 +145,79 @@ func libhoneyConfig() libhoney.ClientConfig {
 		log.Warnln("LIBHONEY_API_KEY or LIBHONEY_DATASET not set, disabling libhoney")
 		return libhoney.ClientConfig{}
 	}
-	userAgent := fmt.Sprintf("honeycomb-lambda-extension-%s/%s", runtime.GOARCH, version)
 
 	return libhoney.ClientConfig{
-		APIKey:  apiKey,
-		Dataset: dataset,
-		APIHost: apiHost,
-		Transmission: &transmission.Honeycomb{
-			MaxBatchSize:          libhoney.DefaultMaxBatchSize,
-			BatchTimeout:          libhoney.DefaultBatchTimeout,
-			MaxConcurrentBatches:  libhoney.DefaultMaxConcurrentBatches,
-			PendingWorkCapacity:   libhoney.DefaultPendingWorkCapacity,
-			UserAgentAddition:     userAgent,
-			EnableMsgpackEncoding: true,
-		},
+		APIKey:       apiKey,
+		Dataset:      dataset,
+		APIHost:      apiHost,
+		Transmission: newTransmission(),
+	}
+}
+
+func newTransmission() *transmission.Honeycomb {
+	batchSendTimeout := envOrElseDurationSeconds("HONEYCOMB_BATCH_SEND_TIMEOUT_S", defaultBatchSendTimeout)
+
+	userAgent := fmt.Sprintf("honeycomb-lambda-extension-%s/%s", runtime.GOARCH, version)
+
+	return &transmission.Honeycomb{
+		MaxBatchSize:          libhoney.DefaultMaxBatchSize,
+		BatchTimeout:          libhoney.DefaultBatchTimeout,
+		MaxConcurrentBatches:  libhoney.DefaultMaxConcurrentBatches,
+		PendingWorkCapacity:   libhoney.DefaultPendingWorkCapacity,
+		UserAgentAddition:     userAgent,
+		EnableMsgpackEncoding: true,
+		BatchSendTimeout:      batchSendTimeout,
 	}
 }
 
-// helper functions for environment variables with default fallbacks
+// (helper) Retrieve an environment variable value by the given key,
+// return an integer based on that value.
+// If key cannot be found or value fails to cast to an int,
+// return the given fallback duration.
 func envOrElseInt(key string, fallback int) int {
 	if value, ok := os.LookupEnv(key); ok {
 		v, err := strconv.Atoi(value)
 		if err != nil {
+			log.Warnf("%s was provided, but failed to parse to an integer. Falling back to default of %d.", key, fallback)
 			return fallback
 		}
 		return v
 	}
 	return fallback
 }
 
+// (helper) Retrieve an environment variable value by the given key,
+// return a boolean based on that value.
+// If key cannot be found or value fails to cast to a bool,
+// return the given fallback duration.
 func envOrElseBool(key string, fallback bool) bool {
 	if value, ok := os.LookupEnv(key); ok {
 		v, err := strconv.ParseBool(value)
 		if err != nil {
+			log.Warnf("%s was provided, but failed to parse to true or false. Falling back to default of %t.", key, fallback)
 			return fallback
 		}
 		return v
 	}
 	return fallback
 }
 
+// (helper) Retrieve an environment variable value by the given key,
+// return a duration in seconds based on that value.
+// If key cannot be found or value fails to cast to an int,
+// return the given fallback duration.
+func envOrElseDurationSeconds(key string, fallback time.Duration) time.Duration {
+	if value, ok := os.LookupEnv(key); ok {
+		v, err := strconv.Atoi(value)
```

Comment:
do you want to allow parsing this as a Duration? Go supports `ParseDuration` and then this can take values like "5s" or "1000ms". If you do, it should probably fall back to checking if it's an undecorated integer as well before giving up.

Reply:
Yes. That could be more better. I'll give that improvement a whirl.

Reply:
Updated in https://github.com/honeycombio/honeycomb-lambda-extension/pull/98/commits/5a46b0e55d0467e278f0a7370b6ea898914b43cb. What do you think? 

![Spongebob Squarepants sitting at a desk in a classroom looking hopeful.](https://user-images.githubusercontent.com/517302/193668347-665fbf55-33a0-471e-9209-b8f2621e2d3d.gif)